### ISSUE TITLE
Consolidate SSL Protocol Settings and Enable TLS v1.2 by Default

### ISSUE THREAD

**Issue by tst-ccamp** [google/earthenterprise#196]
TLS v1.2 is the preferred security protocol for SSL connections. However, the /opt/google/gehttpd/conf.d/security.conf file contains these lines:
```
# Limit SSL protocols.
SSLProtocol -ALL +SSLv3 +TLSv1
# Override SSL configuration to require more secure ciphers.
SSLCipherSuite ALL:!aNULL:!ADH:!eNULL:!LOW:!EXP:RC4+RSA:+HIGH:+MEDIUM
```
It's further confused by the fact that the /opt/google/gehttpd/conf/extra/httpd-ssl.conf also contains these entries with no indication that I have found that they will be overridden elsewhere. 

AC:
1. Make sure that TLSv1.2 is enabled in the default configuration and that it is listed first on the SSLProtocol line so that it will be considered to be preferred
1. Eliminate the confusion caused by having this configuration in two places or at the least make sure that it's obvious that is what is going on
1. Update documentation to explain how to configure the SSL values (see http://www.opengee.org/geedocs/answer/6080928.html)

**Comment by asmitaw** [comment]
little more info on this ticket the TLS v1.2 is not turned ON by default to be compatible with the EC client 
stack that still supports no more than TLS v1.0

see this
https://groups.google.com/forum/#!searchin/google-earth-enterprise/TLS|sort:relevance/google-earth-enterprise/59_uuP9FiwE/vjPNx0JHGQAJ

**Comment by asmitaw** [comment]
I think if we change it, it will create more confusion.

**Comment by tst-ccamp** [comment]
@asmitaw Turning on TLS 1.2 does not also turn off TLS 1.0. The SSL handshake will go through each of the protocols in order until it finds one that works.  Have you seen any issues when you enable TLS 1.2 but still leave TLS 1.0 available?

**Comment by tst-ccamp** [comment]
Also Avnish's suggestion in the thread doesn't work because the setting in httpd-ssl.conf is overridden by the setting in security.conf.

**Comment by asmitaw** [comment]
In apache config /opt/google/gehttpd/conf/gehttpd.conf , following settings indicate that we dont use the default SSL settings ( hence the comment line #Include conf/extra/httpd-ssl.conf ) AND further we have Google Earth specific setting that we use from Include conf.d/*.conf that would also include then,
/opt/google/gehttpd/conf.d/security.conf

So, it makes sense to update /opt/google/gehttpd/conf.d/security.conf only and document this.  I believe we do not need to change httpd-ssl.conf 

# Secure (SSL/TLS) connections
#Include conf/extra/httpd-ssl.conf
#
# Note: The following must must be present to support
#       starting without SSL on platforms with no /dev/random equivalent
#       but a statically compiled-in mod_ssl.
#
<IfModule ssl_module>
SSLRandomSeed startup builtin
SSLRandomSeed connect builtin
</IfModule>

# Include Google Earth Server-specific files
Include conf.d/*.conf

**Comment by tst-ccamp** [comment]
If we're going to stick with keeping this setting in security.conf, we should remove the line in httpd-ssl.conf altogether.  instead of commenting it out. In linux config files it's typical to have commented-out lines that you enable by uncommenting. However, that won't work in this case, so it would be confusing to allow it to remain in the httpd-ssl.conf file.

**Comment by asmitaw** [comment]
[here](https://github.com/google/earthenterprise/blob/7dbd0ded23620ffdbd73f21f3a4d3cf9ed8d807e/earth_enterprise/src/third_party/apache2/apache-ge.spec#L228) is where google earth specific files get included in the gehttpd.conf :


@tst-ccamp : to your point of deleting the comment line, its not a static file, it comes with the apache package with all those commented includes at the end :
earth_enterprise/third_party/apache2/httpd-2.2.32.tar.gz/docs/conf/httpd.conf.in

**Comment by asmitaw** [comment]
More info here to explain the settings (for future reference):
https://httpd.apache.org/docs/2.2/mod/mod_ssl.html

**Comment by tst-ccamp** [comment]
@asmitaw I personally spent at least a couple hours trying to debug why it wouldn't work when I uncommented the line. As a second example, Avnish in the thread you linked to also mistakenly thought you could uncomment the "SSLProtocol" to make changes. If it confused the two of us who have experience with GEE configuration, it will certainly confuse an average user. We simply should not include this configuration parameter in two different places, even if it's commented out in one of them. There is an AC on the ticket to eliminate confusion (it's a vague requirement, admittedly), so this part of the task isn't beyond the scope of the ticket. There are examples of precedence with other libraries included in the fusion build of modifying files after they are extracted.

**Comment by tst-lsavoie** [comment]
Closed via PR #242.
